facebookads/api.py

```diff
@@ -500,25 +500,26 @@ def execute(self):
         retry_indices = []
 
         for index, response in enumerate(responses):
-            if response:
-                body = response.get('body')
-                code = response.get('code')
-                headers = response.get('headers')
-
-                inner_fb_response = FacebookResponse(
-                    body=body,
-                    headers=headers,
-                    http_status=code,
-                    call=self._batch[index],
-                )
-
-                if inner_fb_response.is_success():
-                    if self._success_callbacks[index]:
-                        self._success_callbacks[index](inner_fb_response)
-                elif self._failure_callbacks[index]:
-                    self._failure_callbacks[index](inner_fb_response)
-            else:
+            if not response:
                 retry_indices.append(index)
+                continue
+
+            inner_fb_response = FacebookResponse(
+                body=response.get('body'),
+                headers=response.get('headers'),
+                http_status=response.get('code'),
+                call=self._batch[index],
+            )
+            if inner_fb_response.is_success() and self._success_callbacks[index]:
+                self._success_callbacks[index](inner_fb_response)
+            else:
+                # retry transient errors
+                if inner_fb_response.is_transient():
+                    retry_indices.append(index)
+
+                # execute failure callbacks on error, even if transient
+                if self._failure_callbacks[index]:
```

Comment:
so the reason I thought it would be nice to call the failure callback is because of our workflow, I'll describe it here but I'm pretty sure it's generic:
- we want to do modifications by batch on 20 objects
- we have a failure callback that does something on the objects (in detail, it sets an error on the object for error handling later)
- we want to implement a retry mechanism on transient issue, BUT we still want to mark our object as errored, in order to signal it to the user
- later an async task will retry the transient failures. At the time of the retry (other thread) there is no trace of the previous objects.

Does that make sense? In that case, could we call the failure callbacks with for empty responses?